pkg/restapi/vc/operation/operations.go

```diff
@@ -775,20 +779,88 @@ func (o *Operation) GetRESTHandlers(mode string) ([]Handler, error) {
 		}, nil
 	case "issuer":
 		return []Handler{
-			support.NewHTTPHandler(createCredentialEndpoint, http.MethodPost, o.createCredentialHandler),
+			// profile
 			support.NewHTTPHandler(createProfileEndpoint, http.MethodPost, o.createProfileHandler),
 			support.NewHTTPHandler(getProfileEndpoint, http.MethodGet, o.getProfileHandler),
+
+			// verifiable credential
+			support.NewHTTPHandler(createCredentialEndpoint, http.MethodPost, o.createCredentialHandler),
 			support.NewHTTPHandler(storeCredentialEndpoint, http.MethodPost, o.storeVCHandler),
 			support.NewHTTPHandler(verifyCredentialEndpoint, http.MethodPost, o.verifyCredentialHandler),
 			support.NewHTTPHandler(updateCredentialStatusEndpoint, http.MethodPost, o.updateCredentialStatusHandler),
 			support.NewHTTPHandler(retrieveCredentialEndpoint, http.MethodGet, o.retrieveVCHandler),
 			support.NewHTTPHandler(vcStatusEndpoint, http.MethodGet, o.vcStatus),
+
+			// issuer apis
+			// TODO update trustbloc components to use these APIs instead of above ones
+			support.NewHTTPHandler(issueCredentialPath, http.MethodPost, o.issueCredentialHandler),
 		}, nil
 	default:
 		return nil, fmt.Errorf("invalid operation mode: %s", mode)
 	}
 }
 
+func (o *Operation) issueCredentialHandler(rw http.ResponseWriter, req *http.Request) {
+	// get the request
+	cred := IssueCredentialRequest{}
+
+	err := json.NewDecoder(req.Body).Decode(&cred)
+	if err != nil {
+		o.writeErrorResponse(rw, http.StatusBadRequest, fmt.Sprintf(invalidRequestErrMsg+": %s", err.Error()))
+
+		return
+	}
+
+	// validate the VC
+	validatedCred, _, err := verifiable.NewCredential(cred.Credential)
+	if err != nil {
+		o.writeErrorResponse(rw, http.StatusBadRequest, fmt.Sprintf("failed to validate credential: %s", err.Error()))
+
+		return
+	}
+
+	// Resolve DID and get the public keyID
+	didDoc, err := o.vdri.Resolve(cred.Opts.AssertionMethod)
+	if err != nil {
+		o.writeErrorResponse(rw, http.StatusBadRequest, fmt.Sprintf("failed to resolve DID : %s", err.Error()))
+
+		return
+	}
+
+	var publicKeyID string
+
+	switch {
+	case len(didDoc.PublicKey) > 0:
+		publicKeyID = didDoc.PublicKey[0].ID
+	case len(didDoc.Authentication) > 0:
+		publicKeyID = didDoc.Authentication[0].PublicKey.ID
```

Comment:
we'll probably revisit this, because IIRC DB validates a VC's proof according to its `proofPurpose` and that has some (non-normative for now) rules of its own